cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,522 @@
+    distributions = client.list_distributions()
+    dist_configs = []
+    if 'Items' in distributions['DistributionList']:
+        for dist in distributions['DistributionList']['Items']:
+            temp_object = dict()
+            config = client.get_distribution_config(
+                Id=dist['Id'])
+            temp_object['Config'] = config['DistributionConfig']
+            temp_object['ETag'] = config['ETag']
+            temp_object['Id'] = dist['Id']
+            temp_object['DomainName'] = dist['DomainName']
+            dist_configs.append(temp_object)
+    return dist_configs
+
+
+def get_existing_origin_access_id(client, module):
+    origin_access_ids = client.list_cloud_front_origin_access_identities()
+    origin_access_configs = []
+    if 'Items' in origin_access_ids['CloudFrontOriginAccessIdentityList']:
+        for origin in origin_access_ids['CloudFrontOriginAccessIdentityList']['Items']:
+            temp_object = dict()
+            config = client.get_cloud_front_origin_access_identity_config(
+                Id=origin['Id'])
+            temp_object['Config'] = config['CloudFrontOriginAccessIdentityConfig']
+            temp_object['ETag'] = config['ETag']
+            temp_object['Id'] = origin['Id']
+            origin_access_configs.append(temp_object)
+    return origin_access_configs
+
+
+def get_existing_invalidations(client, module):
+    distribution_id = module.params.get('resource_id')
+    if not distribution_id:
+        module.fail_json(msg="distribution_id is required for invalidations")
+
+    invalidations = client.list_invalidations(DistributionId=distribution_id)
+
+    invalidation_configs = []
+    if 'Items' in invalidations['InvalidationList']:
+        for invalidation in invalidations['InvalidationList']['Items']:
+            temp_object = dict()
+            config = client.get_invalidation(
+                DistributionId=distribution_id, Id=invalidation['Id'])
+            temp_object['Config'] = config['Invalidation']['InvalidationBatch']
+            temp_object['ETag'] = ""
+            temp_object['Id'] = invalidation['Id']
+            invalidation_configs.append(temp_object)
+    return invalidation_configs
+
+
+def get_existing_streaming_distributions(client, module):
+    distributions = client.list_streaming_distributions()
+    dist_configs = []
+    for dist in distributions['StreamingDistributionList']['Items']:
+        temp_object = dict()
+        config = client.get_streaming_distribution_config(
+            Id=dist['Id'])
+        temp_object['Config'] = config['StreamingDistributionConfig']
+        temp_object['ETag'] = config['ETag']
+        temp_object['Id'] = dist['Id']
+        temp_object['DomainName'] = dist['DomainName']
+        dist_configs.append(temp_object)
+    return dist_configs
+
+
+def creation_setup(client, module):
+    policy = None
+    changed = False
+    params = dict()
+
+    if module.params.get('policy_json'):
```

Comment:
I would prefer to see a single option `policy` (or smth like that) and then within the code detect if it's a JSON string or a path and then act accordingly instead of two different (`policy_json` and `policy_document`) options for what amounts to the same thing.